autoemulate/experimental/save.py

```diff
@@ -0,0 +1,155 @@
+import ast
+from pathlib import Path
+
+import joblib
+import pandas as pd
+
+from autoemulate.experimental.emulators.base import Emulator
+from autoemulate.experimental.results import Result  # , Results
+
+
+class ModelSerialiser:
+    def __init__(self, logger):
+        self.logger = logger
+
+    def _save_model(
+        self, model: Emulator, model_name: str | None, path: str | Path | None = None
+    ):
+        """Saves a model to disk.
+
+        Parameters
+        ----------
+        model : Emulator model
+            Model to save.
+        model_name : str or None, optional
+            Name of the model to save.
+            If None, the model's name will be used.
+        path : str, Path or None, optional
+            Path to save the model.
+            If None, the model will be saved with the model name.
+        """
+        if model_name is None:
+            model_name = self._get_model_name(model)
+        full_path = self._prepare_path(path, model_name)
+
+        try:
+            joblib.dump(model, full_path)
+            self.logger.info("%s saved to %s", model_name, full_path)
+            return full_path
+        except Exception as e:
+            self.logger.error("Failed to save %s to %s: %s", model_name, full_path, e)
+            raise
+
+    def _save_result(
+        self, result: Result, result_name: str | None, path: str | Path | None = None
+    ) -> Path:
+        """Saves a model to disk.
```

Comment:
just trying to make it clearer how this method is different from `_save_model`

```suggestion
        """Saves a result and model to disk.
```